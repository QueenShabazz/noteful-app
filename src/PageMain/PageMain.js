import React from 'react';
import FolderNotes from '../FolderNotes/FolderNotes'
import { findNote } from '../notes-helper'
import ApiContext from '../ApiContext'

export default class PageMain extends React.Component{

    static defaultProps ={
        match: {
            params: {}
        }
    }
   
    static contextType = ApiContext


    render(){
    
    const { notes = [] } = this.context
    const note = findNote(notes, this.props.match.params.noteId)
     || { content: '' }
    const noteId =this.props.match.params.noteId
       
        return (
            < section className='PageMain' >
              <FolderNotes
                params = {noteId}
                id={notes.map(id=> id.id)}
                name={notes.map(name=> name.name)}
                content={notes.map(content=> content.content)}
                modified={notes.map(m=> m.modified)}
                onDeleteNote={this.props.onDeleteNote}
                history={this.props.history}
              />
              <div>
                {note.content.split(/\n \r|\n/).map((para, i) =>
                  <p key={i}>{para}</p>
                )}
              </div>
            </section >
          )
        }
    }