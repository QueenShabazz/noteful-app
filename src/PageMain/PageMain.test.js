import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PageMain from './PageMain'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() });

describe(`Form component`, () => {

  it('renders the Form default', () => {
    const wrapper = shallow(
        <PageMain />
        )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
})