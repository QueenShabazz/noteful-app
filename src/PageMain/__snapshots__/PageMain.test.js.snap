// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders the Form default 1`] = `
<section
  className="PageMain"
>
  <FolderNotes
    content={Array []}
    id={Array []}
    modified={Array []}
    name={Array []}
  />
  <div>
    <p
      key="0"
    />
  </div>
</section>
`;
